Firmware/mmu2_reporting.cpp

```diff
@@ -295,30 +295,32 @@ void ReportProgressHook(CommandInProgress cip, ProgressCode ec) {
     }
 }
 
-void TryLoadUnloadProgressbarInit() {
-    lcd_clearstatus();
+TryLoadUnloadReporter::TryLoadUnloadReporter(float delta_mm)
+: dpixel0(0)
+, dpixel1(0)
+, lcd_cursor_col(0)
+, pixel_per_mm(0.5F * float(LCD_WIDTH) / (delta_mm))
+{
+    // Clear the status line
+    lcd_set_cursor(0, 3);
+    lcd_space(LCD_WIDTH);
 }
 
-void TryLoadUnloadProgressbarDeinit() {
-    // Delay the next status message just so
-    // the user can see the results clearly
+void TryLoadUnloadReporter::Render(uint8_t col, bool sensorState) {
+    // Set the cursor position each time in case some other
+    // part of the firmware changes the cursor position
+    lcd_putc_at(col, 3, sensorState ? '-' : LCD_STR_SOLID_BLOCK[0]);
     lcd_reset_status_message_timeout();
```

Comment:
Here we should use `if (!lcd_update_enabled) lcdui_print_status_line();` this is hack to allow showing the progress bar in fullscreens like "Loading Filament 1".